### PR TITLE
testsuite: mailx does not create dead.letter in failure

### PR DESCRIPTION
**Pull request by mhabrnal**
Signed-off-by: Matej Habrnal <mhabrnal@redhat.com>

### CODE REVIEW

File: tests/runtests/mailx-dead-letter/runtest.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        rlRun "mv -f $SENDMAIL ${SENDMAIL}.backup"
+
+        prepare
+        generate_crash
+        get_crash_path
+        wait_for_hooks
+
+        rlRun "ls $crash_PATH > dir_content.log"
+
+        rlRun "reporter-mailx -d $crash_PATH -c mailx.conf" 1
```
Comment (jfilak) [inline tests/runtests/mailx-dead-letter/runtest.sh]: I think `reporter-mailx` creates `reported_to` file if it does not exist.

File: tests/runtests/mailx-dead-letter/runtest.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+# vim: dict=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   runtest.sh of mailx-reporting
+#   Description: Tests abrt-action-mailx
+#   Author: Michal Nowak <mnowak@redhat.com>
```
Comment (jfilak) [inline tests/runtests/mailx-dead-letter/runtest.sh]: You are not Michal Nowak, are you?

File: tests/runtests/mailx-dead-letter/runtest.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+# # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright (c) 2011 Red Hat, Inc. All rights reserved.
```
Comment (jfilak) [inline tests/runtests/mailx-dead-letter/runtest.sh]: 2016

File: tests/runtests/mailx-dead-letter/runtest.sh (hunk 1 above)
Comment (mhabrnal) [inline tests/runtests/mailx-dead-letter/runtest.sh]: It creates ``reported_to`` file only in the case ``reporter-mailx`` sent the mail successfully. ``reporter-mailx``, In this case, should fail, so checking not creating ``reported-to`` file is desirable.

**Pull request merged**